> @anushreeringne Is this spec management plane or data plane?
> 
> For the prettier fix please follow the exact command:
> 
> moba

This is data plane. As I mentioned I did try to run prettier but was unable to fix the errors. 

> @anushreeringne Is this spec management plane or data plane?
> 
> For the prettier fix please follow the exact command:
> 
> npm install; npm run prettier -- --write "specification/hdinsight/**/*.json"

Hi, I got the following error when I tried to run it. I am running it on "Ubuntu 16.04.6 LTS"" and I did update the latest node.js and npm.

npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@2.1.2
npm ERR! Linux 4.15.0-1066-azure
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn

npm ERR! autorest@2.0.4413 preinstall: `node ./preinstall-check`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the autorest@2.0.4413 preinstall script 'node ./preinstall-check'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the autorest package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./preinstall-check
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs autorest
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls autorest
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/sshuser/azure-rest-api-specs/npm-debug.log

> azure-rest-api-specs-tests@0.1.0 prettier /home/sshuser/azure-rest-api-specs
> prettier "--write" "specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json"

sh: prettier: command not found

npm ERR! Linux 4.15.0-1066-azure
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "prettier" "--" "--write" "specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! azure-rest-api-specs-tests@0.1.0 prettier: `prettier "--write" "specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json"`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the azure-rest-api-specs-tests@0.1.0 prettier script 'prettier "--write" "specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json"'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the azure-rest-api-specs-tests package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     prettier "--write" "specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json"
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs azure-rest-api-specs-tests
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls azure-rest-api-specs-tests
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/sshuser/azure-rest-api-specs/npm-debug.log

--- a/specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json
+++ b/specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json
@@ -228,7 +228,7 @@
         "tags": [ "v1metadata" ],
         "summary": "Get metadata about a specific Kafka topic-partition",
         "description": "Get metadata about a specific Kafka topic-partition with partition ID, earliest offset, latest offset, leader, replicas, and in-sync replicas",
-        "operationId": "Metadata_GetMetadataForPartitionId",
+        "operationId": "Metadata_GetMetadataForPartition",
         "consumes": [ "application/json" ],
         "produces": [ "application/json" ],
         "parameters": [

--- a/specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json
+++ b/specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json
@@ -39,7 +39,7 @@
         "tags": [ "v1consumer" ],
         "summary": "Consume records using a simple consumer",
         "description": "Consume records from one partition of a topic beginning with a specific offset. By default count is 1. It can read maximum of 6 megabytes of data at once",
-        "operationId": "consumeTopicInPartitionWithOffset",
+        "operationId": "Consumer_ConsumeTopicInPartitionWithOffset",
         "consumes": [ "application/json" ],
         "produces": [ "application/json" ],
         "parameters": [
@@ -111,7 +111,7 @@
         "tags": [ "v1metadata" ],
         "summary": "Get a list of Kafka brokers",
         "description": "Get a list of Kafka brokers with hostname, port, and broker id",
-        "operationId": "getBrokers",
+        "operationId": "Metadata_GetBrokers",
         "consumes": [ "application/json" ],
         "produces": [ "application/json" ],
         "responses": {
@@ -147,7 +147,7 @@
         "tags": [ "v1metadata" ],
         "summary": "Get a list of Kafka topics",
         "description": "Get a list of Kafka topics. If it is a newly created topic, please wait a few seconds for it to show up.",
-        "operationId": "getTopics",
+        "operationId": "Metadata_GetTopics",
         "consumes": [ "application/json" ],
         "produces": [ "application/json" ],
         "responses": {
@@ -183,7 +183,7 @@
         "tags": [ "v1metadata" ],
         "summary": "Get metadata about all partitions for a specific topic",
         "description": "Get metadata about all partitions for a specific topic with partition ID, earliest offset, latest offset, leader, replicas, and in-sync replicas",
-        "operationId": "getPartitions",
+        "operationId": "Metadata_GetTopicPartitions",
         "consumes": [ "application/json" ],
         "produces": [ "application/json" ],
         "parameters": [
@@ -228,7 +228,7 @@
         "tags": [ "v1metadata" ],
         "summary": "Get metadata about a specific Kafka topic-partition",
         "description": "Get metadata about a specific Kafka topic-partition with partition ID, earliest offset, latest offset, leader, replicas, and in-sync replicas",
-        "operationId": "getPartitionMetadata",
+        "operationId": "Metadata_GetMetadataForPartitionId",
         "consumes": [ "application/json" ],
         "produces": [ "application/json" ],
         "parameters": [
@@ -281,7 +281,7 @@
         "tags": [ "v1producer" ],
         "summary": "Produce records",
         "description": "Produce records to a topic in Kafka. If producing records to a newly created topic, please wait a few seconds for the topic to show up.",
-        "operationId": "produceMessageToTopic",
+        "operationId": "Producer_ProduceMessageToTopic",
         "consumes": [ "application/json" ],
         "produces": [ "application/json" ],
         "parameters": [
@@ -335,7 +335,7 @@
         "tags": [ "v1status" ],
         "summary": "Check the status of Kafka Rest proxy server",
         "description": "Check the status of Kafka Rest proxy server",
-        "operationId": "checkRestproxyStatus",
+        "operationId": "Admin_CheckRestProxyStatus",
         "consumes": [ "application/json" ],
         "produces": [ "application/json" ],
         "responses": {
@@ -353,7 +353,7 @@
         "tags": [ "v1topics" ],
         "summary": "Create a topic",
         "description": "Create a topic in Kafka with topic configuration",
-        "operationId": "putTopics",
+        "operationId": "Admin_CreateTopic",
         "consumes": [ "application/json" ],
         "produces": [ "application/json" ],
         "parameters": [

--- a/specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json
+++ b/specification/hdinsight/hdinsight-kafka-rest-proxy/proxy.json
@@ -32,16 +32,24 @@
       "description": "Manage Kafka topics"
     }
   ],
-  "schemes": [ "https" ],
+  "schemes": [
+    "https"
+  ],
   "paths": {
     "/v1/consumer/topics/{topic}/partitions/{partition}/offsets/{offset}": {
       "get": {
-        "tags": [ "v1consumer" ],
+        "tags": [
+          "v1consumer"
+        ],
         "summary": "Consume records using a simple consumer",
         "description": "Consume records from one partition of a topic beginning with a specific offset. By default count is 1. It can read maximum of 6 megabytes of data at once",
         "operationId": "Consumer_ConsumeTopicInPartitionWithOffset",
-        "consumes": [ "application/json" ],
-        "produces": [ "application/json" ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
             "name": "topic",
@@ -108,12 +116,18 @@
     },
     "/v1/metadata/brokers": {
       "get": {
-        "tags": [ "v1metadata" ],
+        "tags": [
+          "v1metadata"
+        ],
         "summary": "Get a list of Kafka brokers",
         "description": "Get a list of Kafka brokers with hostname, port, and broker id",
         "operationId": "Metadata_GetBrokers",
-        "consumes": [ "application/json" ],
-        "produces": [ "application/json" ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
         "responses": {
           "200": {
             "description": "successful operation",
@@ -144,12 +158,18 @@
     },
     "/v1/metadata/topics": {
       "get": {
-        "tags": [ "v1metadata" ],
+        "tags": [
+          "v1metadata"
+        ],
         "summary": "Get a list of Kafka topics",
         "description": "Get a list of Kafka topics. If it is a newly created topic, please wait a few seconds for it to show up.",
         "operationId": "Metadata_GetTopics",
-        "consumes": [ "application/json" ],
-        "produces": [ "application/json" ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
         "responses": {
           "200": {
             "description": "successful operation",
@@ -180,12 +200,18 @@
     },
     "/v1/metadata/topics/{topic}/partitions": {
       "get": {
-        "tags": [ "v1metadata" ],
+        "tags": [
+          "v1metadata"
+        ],
         "summary": "Get metadata about all partitions for a specific topic",
         "description": "Get metadata about all partitions for a specific topic with partition ID, earliest offset, latest offset, leader, replicas, and in-sync replicas",
         "operationId": "Metadata_GetTopicPartitions",
-        "consumes": [ "application/json" ],
-        "produces": [ "application/json" ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
             "name": "topic",
@@ -225,12 +251,18 @@
     },
     "/v1/metadata/topics/{topic}/partitions/{partition}": {
       "get": {
-        "tags": [ "v1metadata" ],
+        "tags": [
+          "v1metadata"
+        ],
         "summary": "Get metadata about a specific Kafka topic-partition",
         "description": "Get metadata about a specific Kafka topic-partition with partition ID, earliest offset, latest offset, leader, replicas, and in-sync replicas",
         "operationId": "Metadata_GetMetadataForPartition",
-        "consumes": [ "application/json" ],
-        "produces": [ "application/json" ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
             "name": "topic",
@@ -278,12 +310,18 @@
     },
     "/v1/producer/topics/{topic}": {
       "post": {
-        "tags": [ "v1producer" ],
+        "tags": [
+          "v1producer"
+        ],
         "summary": "Produce records",
         "description": "Produce records to a topic in Kafka. If producing records to a newly created topic, please wait a few seconds for the topic to show up.",
         "operationId": "Producer_ProduceMessageToTopic",
-        "consumes": [ "application/json" ],
-        "produces": [ "application/json" ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
             "name": "topic",
@@ -332,12 +370,18 @@
     },
     "/v1/status": {
       "get": {
-        "tags": [ "v1status" ],
+        "tags": [
+          "v1status"
+        ],
         "summary": "Check the status of Kafka Rest proxy server",
         "description": "Check the status of Kafka Rest proxy server",
         "operationId": "Admin_CheckRestProxyStatus",
-        "consumes": [ "application/json" ],
-        "produces": [ "application/json" ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
         "responses": {
           "200": {
             "description": "successful operation",
@@ -350,12 +394,18 @@
     },
     "/v1/topics/{topic}": {
       "put": {
-        "tags": [ "v1topics" ],
+        "tags": [
+          "v1topics"
+        ],
         "summary": "Create a topic",
         "description": "Create a topic in Kafka with topic configuration",
         "operationId": "Admin_CreateTopic",
-        "consumes": [ "application/json" ],
-        "produces": [ "application/json" ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
             "name": "topic",
@@ -447,7 +497,9 @@
     },
     "ProducerRecord": {
       "type": "object",
-      "required": [ "value" ],
+      "required": [
+        "value"
+      ],
       "properties": {
         "value": {
           "type": "object",

